### PR TITLE
Add crew cache option

### PR DESCRIPTION
**Pull request by satmandu**
When developing packages with chromebrew, archive and source packages get downloaded over and over again, wasting bandwidth.

This modification lets you set a ```CREW_CACHE_OPT``` environment variable, which activates logic to both look for and save archives (package and source) during installs.

Not setting the variable leads to current behavior.

Works properly:
- [x] x86_64

### CODE REVIEW

File: crew [hunk 1]
```diff
@@ -497,15 +497,36 @@ def download
...
       system('curl', '--retry', '3', '--progress-bar', '-C', '-', '--insecure', '-L', '-#', url, '-o', filename)
     end
+
     abort 'Checksum mismatch. :/ Try again.'.lightred unless
       Digest::SHA256.hexdigest( File.read("./#{filename}") ) == sha256sum
+    puts "Archive downloaded".lightgreen
+
+    unless ENV["CREW_CACHE_OPT"].to_s == ''
+      puts "Caching downloaded archive".lightgreen
+      FileUtils.cp "#{filename}", "#{CREW_CACHE_DIR}"
```
Comment (uberhacker) [inline crew:527]: This line could be simply:
```
FileUtils.cp filename, CREW_CACHE_DIR
```
No need to enclose in quotes.

**Comment by uberhacker** [comment]
This is a good idea @satmandu.  However, should we also have the ability to delete the cache?  I can see where this could fill up the disk after awhile.

**Comment by satmandu** [comment]
@uberhacker This is why I made it an undocumented non-default option. There's no option to deal with filling up the disk from having too many packages, so I'm not sure this is the best place to start having logic about disk space usage?

**Comment by satmandu** [comment]
(Since it is a hidden option, we could add more hidden options to clear out the folder, but I wonder if that code would just bitrot.)

**Comment by uberhacker** [comment]
Well, you can always remove projects with `crew remove`.  There is no command to clear this cache.  I suppose you could add to clear_cache.rb?

**Comment by satmandu** [comment]
Ah! I could just drop the cache folder into ```~/.cache/crew``` - that way```clear-cache``` wouldn't have to be changed.

File: crew (hunk 1 above)
Comment (satmandu) [reply crew]: done

**Comment by satmandu** [comment]
@uberhacker modified the cache folder to be under home, and now also allowing CREW_CACHE_DIR to be pulled as an env variable too like CREW_PREFIX.

File: lib/const.rb [hunk 2]
```diff
@@ -34,6 +34,14 @@
   HOME = CREW_PREFIX + ENV['HOME']
 end
 
+if ENV['CREW_CACHE_DIR'].to_s == ''
+  CREW_CACHE_DIR = HOME + '.cache/crewcache'
```
Comment (uberhacker) [inline lib/const.rb:37]: Do we need `'/.cache/crewcache'`?

File: lib/const.rb [hunk 3]
```diff
@@ -34,6 +34,14 @@
...
+else
+  CREW_CACHE_DIR = ENV['CREW_CACHE_DIR']
+  @pkg.build_from_source = true
```
Comment (uberhacker) [inline lib/const.rb:40]: Why is this line necessary?

File: lib/const.rb (hunk 2 above)
Comment (satmandu) [reply lib/const.rb:37]: ~/.cache seems like it could be used by multiple programs, so let's not clobber?

File: lib/const.rb (hunk 3 above)
Comment (satmandu) [reply lib/const.rb:40]: oops. good catch.

Comment (satmandu) [reply lib/const.rb]: copypasta fixed

File: lib/const.rb (hunk 2 above)
Comment (satmandu) [reply lib/const.rb:37]: Do we need a ```/``` before ```./cache```?

Comment (uberhacker) [reply lib/const.rb:37]: Yeah, that was my question.

Comment (satmandu) [reply lib/const.rb]: Thanks for getting me there. :)  Fixed.
